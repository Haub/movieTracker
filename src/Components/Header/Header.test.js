import React from 'react';
import { Header } from './index.js';
import { shallow } from 'enzyme';
import { mapDispatchToProps } from './index.js';

describe('HEADER', () => {
  let mockUser;
  let mockActivateLogin;
  let mockSearchMovies;
  
  beforeEach(() => {
    mockUser = {id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom']};
    mockActivateLogin = jest.fn();
    mockSearchMovies = jest.fn();
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<Header user={{ id: 2 }} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot with props', () => {
    const wrapper = shallow(
      <Header 
        user={mockUser}
        activateLogin={mockActivateLogin}
        searchMovies={mockSearchMovies}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls handleSearch when input is changed', () => {
    const wrapper = shallow(
      <Header 
        searchMovies={mockSearchMovies} 
        user={{ id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom'] }}
      />);
    const spy = jest.spyOn(wrapper.instance(), 'handleSearch');
    wrapper.instance().forceUpdate();
    const mockEvent = { target: {value: 'movie'}};
    wrapper.find('.search-field').simulate('change', mockEvent);
    expect(spy).toHaveBeenCalled();

  })
})

describe('mapDispatchToProps', () => {
  it('should call loginUser when logOutUser is invoked', () => {
    const mockDispatch = jest.fn();
    const expected = {"type": "LOGIN_USER", "user": {"email": "graham@aol.com", "favorites": ["Venom"], "id": 2, "name": "Graham", "password": "123"}};
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.loginUser({id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom']});
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  })

})