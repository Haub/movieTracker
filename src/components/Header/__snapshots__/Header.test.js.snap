// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER should match the snapshot 1`] = `
<header>
  <Link
    replace={false}
    to="/"
  >
    <img
      alt="home logo"
      className="logo"
      src="logo.svg"
    />
  </Link>
  <div
    className="controls-container"
  >
    <form
      className="search-form"
      role="search"
    >
      <label>
        <input
          autoComplete="off"
          className="search-field"
          onChange={[Function]}
          placeholder="Search movies"
          style={
            Object {
              "backgroundImage": "url(search-icon.svg)",
            }
          }
          type="search"
        />
      </label>
      <input
        className="search-submit"
        type="submit"
      />
    </form>
    <img
      alt="user tile"
      className="tile"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "superhero-c.svg",
        }
      }
    />
  </div>
</header>
`;

exports[`HEADER should match the snapshot with props 1`] = `
<header>
  <Link
    replace={false}
    to="/"
  >
    <img
      alt="home logo"
      className="logo"
      src="logo.svg"
    />
  </Link>
  <div
    className="controls-container"
  >
    <form
      className="search-form"
      role="search"
    >
      <label>
        <input
          autoComplete="off"
          className="search-field"
          onChange={[Function]}
          placeholder="Search movies"
          style={
            Object {
              "backgroundImage": "url(search-icon.svg)",
            }
          }
          type="search"
        />
      </label>
      <input
        className="search-submit"
        type="submit"
      />
    </form>
    <img
      alt="user tile"
      className="tile"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "superhero-a.svg",
        }
      }
    />
  </div>
</header>
`;
